src/plugin/declarativedbusobject.cpp

```diff
@@ -0,0 +1,287 @@
+/****************************************************************************************
+**
+** Copyright (c) 2021 Jolla Ltd.
+** All rights reserved.
+**
+** You may use this file under the terms of the GNU Lesser General
+** Public License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file license.lgpl included in the packaging
+** of this file.
+**
+** This library is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+** Lesser General Public License for more details.
+**
+****************************************************************************************/
+
+#include "declarativedbusobject.h"
+
+#include "declarativedbusadaptor.h"
+
+/*!
+    \qmltype DBusObject
+    \inqmlmodule Nemo.DBus
+    \brief Provides a service on D-Bus
+
+    The DBusObject object can be used to register a D-Bus object implementing multiple interfaces
+    on the system or session bus.
+
+    Each interface implemented by the object is defined in an individual DBusAdaptor in the body
+    of the DBusObject and will be registered as belonging to the same path and the same optional
```

Comment:
It's not either uncommon that one service has multiple objects in different paths, or that different interfaces have different object paths. We can have support only for this variation first, but should have some plan on going forward.

What do you think about that case? DBusService { service: "foo.bar"; DBusObject { path: "/"; DBusAdaptor { ... } } or something like that, optionally the object part skpiped?

Reply:
We have applications that mix QML dbus objects with C++ objects, some instantiated in the QML application, some in main or in some other construct.

Having the only dbus object in your application register the service for you is a convenience that can't really be rolled back now. But if you have multiple objects for a single service the solution that handles every combination is to not have an object that is responsible for registering the service for that application but to register it yourself in main after you've constructed everything.